### PR TITLE
Grafana deployment Jenkinsfile 

### PR DESCRIPTION
**Pull request by ebattat**
## Type of change
Note: Fill **x** in []
- [ ] bug
- [x] enhancement
- [ ] documentation
- [ ] dependencies

## Description

This Jenkinsfile is responsible for generating Grafonnet (Jsonnet) Grafana metadata code and updating new versions automatically in the GitHub repository

### CODE REVIEW

File: jenkins/PerfCI_Grafana_Deployment/Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+    stages {
+        stage('Cleanup') {
+            steps {
+                script {
+                        // Clean the Jenkins workspace
+                        echo "Cleaning Jenkins workspace"
+                        deleteDir()
+                        //sh '[ -d ${WORKSPACE} ] && rm -rf ${WORKSPACE}'
+                    sh '''
+                        if [[ "$(sudo podman images -q quay.io/ebattat/benchmark-runner 2> /dev/null)" != "" ]]; then
```
Comment (RobertKrawitz) [inline jenkins/PerfCI_Grafana_Deployment/Jenkinsfile:29]: Another way to write this is

```
if [[ -n "$(sudo podman images -q quay.io/ebattat/benchmark-runner 2> /dev/null)" ]] ; then
...
fi
```
As written it isn't wrong, just a reference for the future.

File: jenkins/PerfCI_Grafana_Deployment/Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+                }
+            }
+        }
+
+       stage('Install dependencies') {
+            steps {
+                script {
+                    sh 'sudo python3 -m pip install --upgrade pip'
+                    sh 'if [[ -f requirements.txt ]]; then pip install -r requirements.txt; fi'
+                    sh 'pip install --upgrade benchmark-runner'
```
Comment (RobertKrawitz) [inline jenkins/PerfCI_Grafana_Deployment/Jenkinsfile:51]: You could probably write this as a single command.

File: jenkins/PerfCI_Grafana_Deployment/Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+        failure {
+            script {
+                msg = "Build error for ${env.JOB_NAME} ${env.BUILD_NUMBER} (${env.BUILD_URL})"
+                emailext body: """\
+                    Jenkins job: ${env.BUILD_URL}\nSee the console output for more details:  ${env.BUILD_URL}consoleFull\n\n
+                """, subject: msg, to: "${CONTACT_EMAIL1}"
+            }
+        }
+        success {
+            echo 'TBD'
```
Comment (RobertKrawitz) [inline jenkins/PerfCI_Grafana_Deployment/Jenkinsfile:164]: Why "TBD"?

**Review by RobertKrawitz — APPROVED** [review]

File: jenkins/PerfCI_Grafana_Deployment/Jenkinsfile (hunk 2 above)
Comment (ebattat) [reply jenkins/PerfCI_Grafana_Deployment/Jenkinsfile:51]: I prefere to keep each command separately and not mix between it.

File: jenkins/PerfCI_Grafana_Deployment/Jenkinsfile (hunk 3 above)
Comment (ebattat) [reply jenkins/PerfCI_Grafana_Deployment/Jenkinsfile:164]: To be defined, it's a placeholder for a successful job. I can remove it if it doesn't look good.

**Pull request merged**